pkg/dns/aws/dns_test.go

```diff
@@ -0,0 +1,218 @@
+	var responses []interface{}
+
+	for _, target := range record.Spec.Targets {
+		elbDescription := &elb.LoadBalancerDescription{
+			DNSName:                   aws.String(target),
+			CanonicalHostedZoneNameID: aws.String(dnsZone.ID),
+		}
+		elbDescriptions = append(elbDescriptions, elbDescription)
+
+		elbResponse := &elb.DescribeLoadBalancersOutput{
+			LoadBalancerDescriptions: elbDescriptions,
+		}
+
+		responses = append(responses, elbResponse)
+	}
+
+	route53Response := &route53.ChangeResourceRecordSetsInput{}
+	responses = append(responses, route53Response)
+
+	return responses
+}
+
+func FakeAWSProvider(resps []interface{}) *Provider {
+	reqNum := 0
+	sess := session.Must(session.NewSession(&aws.Config{
+		DisableSSL: aws.Bool(true),
+	}))
+	sess.Handlers.Send.Clear()
+	sess.Handlers.Unmarshal.Clear()
+	sess.Handlers.UnmarshalMeta.Clear()
+	sess.Handlers.ValidateResponse.Clear()
+	sess.Handlers.Unmarshal.PushBack(func(r *request.Request) {
+		r.Data = resps[reqNum]
+		reqNum++
+	})
+
+	config := &aws.Config{
+		MaxRetries: aws.Int(1),
+		Region:     aws.String("ap-southeast-2"),
+	}
+
+	route53Config := sess.ClientConfig("route53", config)
+	route53 := &route53.Route53{
+		Client: client.New(
+			*route53Config.Config,
+			metadata.ClientInfo{
+				ServiceName:   "route53",
+				ServiceID:     "Route 53",
+				SigningName:   "",
+				SigningRegion: route53Config.SigningRegion,
+				Endpoint:      route53Config.Endpoint + "/route53",
+				APIVersion:    "2013-04-01",
+				JSONVersion:   "1.1",
+				TargetPrefix:  "MockServer",
+			},
+			route53Config.Handlers,
+		),
+	}
+
+	elbConfig := sess.ClientConfig("elasticloadbalancing", config)
+	elb := &elb.ELB{
+		Client: client.New(
+			*elbConfig.Config,
+			metadata.ClientInfo{
+				ServiceName:   "elasticloadbalancing",
+				ServiceID:     "Elastic Load Balancing",
+				SigningName:   "",
+				SigningRegion: elbConfig.SigningRegion,
+				Endpoint:      elbConfig.Endpoint + "/elb",
+				APIVersion:    "2012-06-01",
+				JSONVersion:   "1.1",
+				TargetPrefix:  "MockServer",
+			},
+			elbConfig.Handlers,
+		),
+	}
+
+	taggingConfig := sess.ClientConfig("resourcegroupstaggingapi", config)
+	tagging := &resourcegroupstaggingapi.ResourceGroupsTaggingAPI{
+		Client: client.New(
+			*taggingConfig.Config,
+			metadata.ClientInfo{
+				ServiceName:   "tagging",
+				ServiceID:     "Resource Groups Tagging API",
+				SigningName:   "",
+				SigningRegion: taggingConfig.SigningRegion,
+				Endpoint:      taggingConfig.Endpoint + "/tagging",
+				APIVersion:    "2017-01-26",
+				JSONVersion:   "1.1",
+				TargetPrefix:  "ResourceGroupsTaggingAPI_20170126",
+			},
+			taggingConfig.Handlers,
+		),
+	}
+
+	return &Provider{
+		elb:            elb,
+		route53:        route53,
+		tags:           tagging,
+		idsToTags:      map[string]map[string]string{},
+		lbZones:        map[string]string{},
+		updatedRecords: sets.NewString(),
+	}
+}
+
+func TestEnsureAWSDNS(t *testing.T) {
+
```

Comment:
Please delete the blank line here (line 129) and the similar blank lines below (lines 152, 175, and 198).